### PR TITLE
sevices/friendbot: Allow friendbot to include cloudflare derived IP address in request logs

### PR DESCRIPTION
**Pull request by tamirms**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://developers.stellar.org/api/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Friendbot uses a logging middleware which is used to log incoming requests. Those logs include the client's IP address. However, that ip address is incorrect if the friendbot http server is load balanced by cloudflare. This PR implements a configuration option which enables friendbot to include the real client ip derived from cloudflare in the request logs.

### Why

It is useful to see real ips in the logs so we can see observe traffic patterns

### Known limitations

[N/A]

### CODE REVIEW

File: services/friendbot/main.go [hunk 1]
```diff
@@ -84,8 +86,12 @@ func run(cmd *cobra.Command, args []string) {
 	})
 }
 
-func initRouter(fb *internal.Bot) *chi.Mux {
-	mux := http.NewAPIMux(log.DefaultLogger)
+func initRouter(cfg Config, fb *internal.Bot) *chi.Mux {
+	mux := chi.NewRouter()
+	// first apply XFFMiddleware so we can have the real ip in the subsequent
+	// middlewares
+	mux.Use(http.XFFMiddleware(http.XFFMiddlewareConfig{BehindCloudflare: cfg.UseCloudflareIP}))
```
Comment (sreuland) [inline services/friendbot/main.go:93]: this is fine for now to get unstuck, but could this be done in a more general non-vendor way to configure getting source ip addresses through a configurable http header name instead? it usually resolves down to an http header, but different name specific to vendor on the lb/firewall, etc.

**Review by sreuland — APPROVED** [review]
are there any integration tests in the repo that bring up friendbot to confirm this config change works?

File: services/friendbot/main.go (hunk 1 above)
Comment (tamirms) [reply services/friendbot/main.go:93]: yes, I think this could be a future improvement for the XFFMiddleware

**Comment by tamirms** [comment]
@sreuland I added a unit test. Does that look good to you?

**Pull request merged**